library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -425,6 +425,13 @@ libspdm_return_t libspdm_set_data(void *spdm_context, libspdm_data_type_t data_t
                          sizeof(context->connection_info.peer_used_cert_chain[slot_id].buffer),
                          data, data_size);
 #else
+        /*process the SPDM cert header and hash*/
+        data = (uint8_t *)data + sizeof(spdm_cert_chain_t) +
+               libspdm_get_hash_size(context->connection_info.algorithm.base_hash_algo);
+        data_size = data_size -
+                    (sizeof(spdm_cert_chain_t) +
+                     libspdm_get_hash_size(context->connection_info.algorithm.base_hash_algo));
```

Comment:
Why we need skip header before hash?

That violates the SPDM spec.

Reply:
I have updated the code. And the unit_test is added to check the code. Thanks.